### PR TITLE
Make compatible with latest qt

### LINKED ISSUE #33: destroyScreen, screenAdded removed from Qt 5.12
It appears that screenAdded and destroyAdded are no longer a thing

```
/tmp/qtmir/src/platforms/mirserver/mirserverintegration.cpp: In lambda function:
/tmp/qtmir/src/platforms/mirserver/mirserverintegration.cpp:160:15: error: ‘class MirServerIntegration’ has no member named ‘destroyScreen’
  160 |         this->destroyScreen(screen);
      |               ^~~~~~~~~~~~~
/tmp/qtmir/src/platforms/mirserver/mirserverintegration.cpp: In member function ‘virtual void MirServerIntegration::initialize()’:
/tmp/qtmir/src/platforms/mirserver/mirserverintegration.cpp:165:9: error: ‘screenAdded’ was not declared in this scope
  165 |         screenAdded(screen);
      |         ^~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by vanyasem**
screenAdded and destroyScreen deprecated

fixes #33

### CODE REVIEW

**Comment by UniversalSuperBox** [comment]
Urgh, it's awful when this happens, but this is one of those cases where you might have to do an `#if QT_VERSION <> QT_VERSION_CHECK()`: https://doc.qt.io/qt-5/qtglobal.html#QT_VERSION_CHECK.

Since both calls have the same interface, you may want to:

1. Create a new function for each of the calls
1. Place the `#if` compiler flags there
1. Add a `# TODO: remove this after we no longer support Qt 5.x` comment above both.

At least, that's the path I'd take. @dobey is better at styling C++ than I am, though. He might have a different opinion.

**Comment by dobey** [comment]
> He might have a different opinion.

We'll just need `#if` checks using `QT_VERSION_CHECK()`  for this. It's how Qt handles such things internally too.

**Comment by vanyasem** [comment]
got it, will add the checks. thank you

**Comment by vanyasem** [comment]
let's see if that works. i introduced 2 functions

**Comment by vanyasem** [comment]
let's see if that works. i introduced 2 functions. Also @UniversalSuperBox comments in CPP start with `//` :P I blindly copied your TODO statement

**Comment by vanyasem** [comment]
yep, apparently the last one worked. @dobey could you please review my changes?

**Review by mariogrip — APPROVED** [review]
LGTM :+1:

File: src/platforms/mirserver/mirserverintegration.cpp [hunk 1]
```diff
@@ -141,6 +141,26 @@ QAbstractEventDispatcher *MirServerIntegration::createEventDispatcher() const
     return createUnixEventDispatcher();
 }
 
+void MirServerIntegration::handleScreenAdded(QPlatformScreen *screen)
+{
+    // TODO: remove this after we no longer support Qt 5.13
```
Comment (dobey) [inline src/platforms/mirserver/mirserverintegration.cpp:146]: I presume you actually mean after we no longer support Qt < 5.13 here?

Comment (UniversalSuperBox) [reply src/platforms/mirserver/mirserverintegration.cpp:146]: Fixed by https://github.com/ubports/qtmir/commit/418fd02e9f683569aed73d01f5c67b6e51cb53f0

**Pull request merged**